diff -Naur scratchbox2-2.3.90.old/modes/obs-rpm-build/fs_rules.lua scratchbox2-2.3.90/modes/obs-rpm-build/fs_rules.lua
--- scratchbox2-2.3.90.old/modes/obs-rpm-build/fs_rules.lua	2013-02-06 10:26:57.000000000 +0200
+++ scratchbox2-2.3.90/modes/obs-rpm-build/fs_rules.lua	2013-03-20 15:24:46.324483508 +0200
@@ -361,6 +361,10 @@
 		 actions=accelerated_program_actions},
 		{path="/usr/bin/yes",
 		 actions=accelerated_program_actions},
+		 
+		{path="/usr/bin/localedef",
+		 func_class = FUNC_CLASS_EXEC,
+		 actions=accelerated_program_actions},
 
 		{path = "/usr/bin/sb2-show", use_orig_path = true,
 		 protection = readonly_fs_always},
diff -Naur scratchbox2-2.3.90.old/modes/obs-rpm-build+pp/fs_rules.lua scratchbox2-2.3.90/modes/obs-rpm-build+pp/fs_rules.lua
--- scratchbox2-2.3.90.old/modes/obs-rpm-build+pp/fs_rules.lua	2013-02-06 10:26:57.000000000 +0200
+++ scratchbox2-2.3.90/modes/obs-rpm-build+pp/fs_rules.lua	2013-03-20 15:26:16.128480137 +0200
@@ -431,6 +431,10 @@
 		{path="/usr/bin/yes",
 		 actions=accelerated_program_actions},
 
+		{path="/usr/bin/localedef",
+		 func_class = FUNC_CLASS_EXEC,
+		 actions=accelerated_program_actions},
+
 		-- perl & python:
 		-- 	processing depends on SBOX_REDIRECT_IGNORE,
 		--	SBOX_REDIRECT_FORCE and 
