From fe9dfbd7b6502ee99d713c3343a6ce204dfb5d12 Mon Sep 17 00:00:00 2001
From: David Greaves <david@dgreaves.com>
Date: Thu, 25 Apr 2013 10:38:08 +0000
Subject: [PATCH] scratchbox2-2.3.52-wrapperargs

---
 wrappers/gdb | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/wrappers/gdb b/wrappers/gdb
index 44cae21..0fc1db5 100755
--- a/wrappers/gdb
+++ b/wrappers/gdb
@@ -5,7 +5,6 @@
 #
 # Licensed under GPL version 2
 
-args="$*"
 prog="$0"
 progbase=`basename $0`
 
@@ -81,4 +80,4 @@ if [ "$sbox_mapmode" = "emulate" ]
 then
 	SHELL=/bin/sh
 fi
-exec $sbox_target_root/usr/bin/gdb -command $SBOX_SESSION_DIR/gdbinit $args
+exec $sbox_target_root/usr/bin/gdb -command $SBOX_SESSION_DIR/gdbinit "$@"
-- 
1.8.3-rc3

