From 8f31c7903590731ccfea5e1d81a7cafa52ec8ad6 Mon Sep 17 00:00:00 2001
From: David Greaves <david@dgreaves.com>
Date: Thu, 25 Apr 2013 10:38:08 +0000
Subject: [PATCH] support-ccache

---
 utils/sb2 |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/utils/sb2 b/utils/sb2
index 87b65b4..5b0f5b0 100755
--- a/utils/sb2
+++ b/utils/sb2
@@ -430,7 +430,11 @@ function sboxify_environment()
 		# /sb2/wrappers and /sb2/scripts contain utilities for
 		# development support, and are available in the
 		# software development modes ("simple" and "devel")
-		export PATH=/sb2/wrappers:/sb2/scripts:$HOME/.scratchbox2/$SBOX_TARGET/bin:$SBOX_DIR/bin:$PATH:/sbin:/usr/sbin:$SBOX_TARGET_ROOT/bin:$SBOX_TARGET_ROOT/usr/bin:$SBOX_TARGET_ROOT/usr/local/bin
+	    TMP_PATH=/sb2/wrappers:/sb2/scripts:$HOME/.scratchbox2/$SBOX_TARGET/bin
+	    if [ ! -z "$CCACHE_TOOLS_PATH" ]; then
+		TMP_PATH=$TMP_PATH:$CCACHE_TOOLS_PATH
+	    fi
+	    export PATH=$TMP_PATH:$SBOX_DIR/bin:$PATH:/sbin:/usr/sbin:$SBOX_TARGET_ROOT/bin:$SBOX_TARGET_ROOT/usr/bin:$SBOX_TARGET_ROOT/usr/local/bin
 	else
 		export PATH=$SBOX_MODE_PATH
 	fi
-- 
1.7.10.3

