From feb9fbb4d0defa2f3d635b8d5f170fca79fbd6bd Mon Sep 17 00:00:00 2001
From: Juha Kallioinen <juha.kallioinen@jollamobile.com>
Date: Thu, 9 May 2013 14:03:21 +0000
Subject: [PATCH] Improve ruletree printout

Signed-off-by: Juha Kallioinen <juha.kallioinen@jollamobile.com>
---
 utils/sb2-ruletree.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/utils/sb2-ruletree.c b/utils/sb2-ruletree.c
index 68d2d8a..88d9cd6 100644
--- a/utils/sb2-ruletree.c
+++ b/utils/sb2-ruletree.c
@@ -278,6 +278,7 @@ static void dump_rules(ruletree_object_offset_t offs, int indent)
 	case SB2_RULETREE_FSRULE_ACTION_IF_EXISTS_IN:
 		printf("if_exists_in '%s'\n",
 			offset_to_ruletree_string_ptr(rule->rtree_fsr_action_offs, NULL));
+		rule_list_link_label = "then_actions";
 		break;
 	default:
 		printf("ERROR: Unknown action type %d\n",
-- 
1.8.3-rc3

