From 61936eacce799f6dcac73588cc2dcc24dd81ee22 Mon Sep 17 00:00:00 2001
From: Juha Kallioinen <juha.kallioinen@jollamobile.com>
Date: Thu, 9 May 2013 20:20:28 +0000
Subject: [PATCH] Exit more gracefully if valid sb2 target not found

Signed-off-by: Juha Kallioinen <juha.kallioinen@jollamobile.com>
---
 utils/sb2 |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/utils/sb2 b/utils/sb2
index b2db500..acf54af 100755
--- a/utils/sb2
+++ b/utils/sb2
@@ -1175,6 +1175,9 @@ END
 		if [ -n "$SBOX_CLONED_TARGET_ROOT" ]; then
 			target_basename=`basename $SBOX_CLONED_TARGET_ROOT`
 		else
+			if [ -z $SBOX_TARGET_ROOT ]; then
+			    exit_error "SBOX_TARGET_ROOT empty, valid target not found"
+			fi
 			target_basename=`basename $SBOX_TARGET_ROOT`
 		fi
 
-- 
1.7.10.3

