From c8a15205c3502bfe3c108c5e50e59ec28a3e6e34 Mon Sep 17 00:00:00 2001
From: Juha Kallioinen <juha.kallioinen@jollamobile.com>
Date: Fri, 10 May 2013 13:55:23 +0000
Subject: [PATCH] Finalize IF_EXISTS_IN forward rule

Signed-off-by: Juha Kallioinen <juha.kallioinen@jollamobile.com>
---
 pathmapping/paths_ruletree_mapping.c |   25 +++++++++++++++++++------
 1 file changed, 19 insertions(+), 6 deletions(-)

diff --git a/pathmapping/paths_ruletree_mapping.c b/pathmapping/paths_ruletree_mapping.c
index 5b0ef1a..24e41ae 100644
--- a/pathmapping/paths_ruletree_mapping.c
+++ b/pathmapping/paths_ruletree_mapping.c
@@ -420,7 +420,7 @@ static int if_exists_in(ruletree_fsrule_t *action,
 		SB_LOG(SB_LOGLEVEL_DEBUG,
 			"if_exists_in: True '%s' -> proceed to then_actions", test_path);
                 free(test_path);
-                return (0);
+                return (1);
 	}
 	SB_LOG(SB_LOGLEVEL_DEBUG,
 		"if_exists_in: False '%s'", test_path);
@@ -662,12 +662,25 @@ static char *ruletree_execute_conditional_actions(
 
                                 case SB2_RULETREE_FSRULE_CONDITION_IF_EXISTS_IN:
                                   if (if_exists_in(action_cand_p, abs_clean_virtual_path)) {
-                                    /* found, continue rule tree processing */
-                                    continue;
-                                  }
-                                  else {
-                                    /* not found */
+                                    /* found, jump to the new rule tree branch */
+                                    ruletree_object_offset_t then_actions_offset = action_cand_p->rtree_fsr_rule_list_link;
+                                    if (!then_actions_offset) {
+                                      SB_LOG(SB_LOGLEVEL_DEBUG, "if_exists_in: no then_actions found");
+                                      /* continue with normal rule tree */
+                                      continue;
+                                    }
+                                    else {
+                                      SB_LOG(SB_LOGLEVEL_DEBUG, "if_exists_in: then_actions found (%d)", then_actions_offset);
+                                      ruletree_fsrule_t new_rules = *rule_selector;
+                                      new_rules.rtree_fsr_rule_list_link = then_actions_offset;
+                                      return ruletree_execute_conditional_actions(ctx, result_log_level,
+                                                                                  abs_clean_virtual_path, flagsp,
+                                                                                  exec_policy_name_ptr, errormsgp,
+                                                                                  &new_rules);
+                                    }
                                   }
+                                  /* not found, continue with normal rule tree */
+                                  continue;
                                   break;
 
 				default:
-- 
1.7.10.3

