From ce9212febf6bf8c2e137868946eadc2425081d01 Mon Sep 17 00:00:00 2001
From: Juha Kallioinen <juha.kallioinen@jollamobile.com>
Date: Thu, 12 Sep 2013 13:57:13 +0300
Subject: [PATCH] [sb2] Exit if target root is not accessible

Signed-off-by: Juha Kallioinen <juha.kallioinen@jollamobile.com>
---
 utils/sb2        | 5 +++++
 utils/sb2-show.c | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/utils/sb2 b/utils/sb2
index acf54af..b2153cd 100755
--- a/utils/sb2
+++ b/utils/sb2
@@ -219,6 +219,11 @@ function load_configuration()
 
 	saved_LD_LIBRARY_PATH=$LD_LIBRARY_PATH
 	. ~/.scratchbox2/$SBOX_TARGET/sb2.config
+
+	if [ ! -d "$SBOX_TARGET_ROOT" ]; then
+	    exit_error "Cannot access target root '$SBOX_TARGET_ROOT', aborting."
+	fi
+
 	LD_LIBRARY_PATH=$saved_LD_LIBRARY_PATH
 
 	# resolve possible symlinks in SBOX_TARGET_ROOT and SBOX_TOOLS_ROOT
diff --git a/utils/sb2-show.c b/utils/sb2-show.c
index 6ad3e2f..91d4efb 100644
--- a/utils/sb2-show.c
+++ b/utils/sb2-show.c
@@ -184,7 +184,7 @@ static void usage_exit(const char *progname, const char *errmsg, int exitstatus,
 	}
 
 	fprintf(stderr, "\n"
-	    "'%s' must be executed inside sb2 sandbox (see the 'sb2'"
+	    "'%s' must be executed inside sb2 sandbox (see the 'sb2' "
 	    "command)\n", progname);
 
 	exit(exitstatus);
-- 
1.8.3-rc3

