diff -ru scratchbox2-2.3.24-nowerror/sb2d/Makefile scratchbox2-2.3.24/sb2d/Makefile
--- scratchbox2-2.3.24-nowerror/sb2d/Makefile	2012-01-28 23:16:48.000000000 +0200
+++ scratchbox2-2.3.24/sb2d/Makefile	2012-01-28 23:28:05.554002066 +0200
@@ -1,7 +1,7 @@
 # beware of namespace collisions, all subdir makefiles are included
 # into the top-level Makefile
 
-$(D)/sb2d: CFLAGS := $(CFLAGS) -Wall -W -Werror \
+$(D)/sb2d: CFLAGS := $(CFLAGS) -Wall -W \
 		$(PROTOTYPEWARNINGS) \
 		-I$(SRCDIR)/include \
 		-I$(SRCDIR)/preload -Ipreload/
Only in scratchbox2-2.3.24/sb2d: Makefile~
diff -ru scratchbox2-2.3.24-nowerror/utils/Makefile scratchbox2-2.3.24/utils/Makefile
--- scratchbox2-2.3.24-nowerror/utils/Makefile	2012-01-11 15:09:24.000000000 +0200
+++ scratchbox2-2.3.24/utils/Makefile	2012-01-28 23:27:55.594261149 +0200
@@ -1,7 +1,7 @@
 # beware of namespace collisions, all subdir makefiles are included
 # into the top-level Makefile
 
-$(D)/sb2-show: CFLAGS := $(CFLAGS) -Wall -W -Werror \
+$(D)/sb2-show: CFLAGS := $(CFLAGS) -Wall -W \
 		-I$(SRCDIR)/preload -Ipreload/ $(PROTOTYPEWARNINGS) \
 		-I$(SRCDIR)/include
 
@@ -11,7 +11,7 @@
 	$(P)LD
 	$(Q)$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ -ldl
 
-$(D)/sb2-ruletree: CFLAGS := $(CFLAGS) -Wall -W -Werror \
+$(D)/sb2-ruletree: CFLAGS := $(CFLAGS) -Wall -W \
 		-I$(SRCDIR)/preload -Ipreload/ $(PROTOTYPEWARNINGS) \
 		-I$(SRCDIR)/include -ldl
 
@@ -21,7 +21,7 @@
 	$(P)LD
 	$(Q)$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
 
-$(D)/sb2-monitor: CFLAGS := $(CFLAGS) -Wall -W -Werror \
+$(D)/sb2-monitor: CFLAGS := $(CFLAGS) -Wall -W  \
 		-I$(SRCDIR)/preload -Ipreload/ $(PROTOTYPEWARNINGS) \
 		-I$(SRCDIR)/include
 
@@ -30,7 +30,7 @@
 	$(P)LD
 	$(Q)$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
 
-$(D)/sb2-interp-wrapper: CFLAGS := $(CFLAGS) -Wall -W -Werror \
+$(D)/sb2-interp-wrapper: CFLAGS := $(CFLAGS) -Wall -W  \
 		-I$(SRCDIR)/preload -Ipreload/ $(PROTOTYPEWARNINGS) \
 		-I$(SRCDIR)/include
 
