diff -ru scratchbox2-2.3.27/modes/obs-rpm-build+pp/exec_rules.lua scratchbox2-2.3.27-fixperl/modes/obs-rpm-build+pp/exec_rules.lua
--- scratchbox2-2.3.27/modes/obs-rpm-build+pp/exec_rules.lua	2012-02-02 08:29:21.000000000 +0100
+++ scratchbox2-2.3.27-fixperl/modes/obs-rpm-build+pp/exec_rules.lua	2012-02-08 07:13:39.870853468 +0100
@@ -194,8 +194,6 @@
 		 exec_policy_name = "Tools-perl"},
 		{prefix = tools_prefix .. "/usr/bin/python",
 		 exec_policy_name = "Tools-python"},
-		-- Rule isn't active if tools_root is not set.
-		{prefix = tools_root, exec_policy_name = "Tools"},
 
                 -- the toolchain, if not from Tools:
                 {dir = sbox_target_toolchain_dir, exec_policy_name = "Toolchain"},
@@ -203,6 +201,9 @@
                 -- the home directory is expected to contain target binaries:
                 {dir = sbox_user_home_dir, exec_policy_name = "Target"},
 
+		-- Rule isn't active if tools_root is not set.
+		{prefix = tools_root, exec_policy_name = "Tools"},
+
 		-- DEFAULT RULE (must exist):
 		{prefix = "/", exec_policy_name = "Host"}
 }
Only in scratchbox2-2.3.27-fixperl/modes/obs-rpm-build+pp: exec_rules.lua~
