diff -ru scratchbox2-2.3.27/modes/obs-rpm-build/fs_rules.lua scratchbox2-2.3.27-usrsrc/modes/obs-rpm-build/fs_rules.lua
--- scratchbox2-2.3.27/modes/obs-rpm-build/fs_rules.lua	2012-02-02 08:29:21.000000000 +0100
+++ scratchbox2-2.3.27-usrsrc/modes/obs-rpm-build/fs_rules.lua	2012-04-02 17:06:05.195777985 +0200
@@ -398,8 +398,6 @@
 
 		{dir = "/usr/lib/gcc", actions = test_first_tools_then_target_default_is_tools},
 
-		{dir = "/usr/src", use_orig_path = true},
-
 		{dir = "/usr", map_to = target_root,
 		protection = readonly_fs_if_not_root}
 }
Only in scratchbox2-2.3.27-usrsrc/modes/obs-rpm-build: fs_rules.lua~
diff -ru scratchbox2-2.3.27/modes/obs-rpm-build+pp/fs_rules.lua scratchbox2-2.3.27-usrsrc/modes/obs-rpm-build+pp/fs_rules.lua
--- scratchbox2-2.3.27/modes/obs-rpm-build+pp/fs_rules.lua	2012-02-02 08:29:21.000000000 +0100
+++ scratchbox2-2.3.27-usrsrc/modes/obs-rpm-build+pp/fs_rules.lua	2012-04-02 17:06:11.704643768 +0200
@@ -538,8 +538,6 @@
 
 		{dir = "/usr/share", rules = usr_share_rules},
 
-		{dir = "/usr/src", use_orig_path = true},
-
 		{dir = "/usr", map_to = target_root,
 		protection = readonly_fs_if_not_root}
 }
Only in scratchbox2-2.3.27-usrsrc/modes/obs-rpm-build+pp: fs_rules.lua~
